### ISSUE TITLE
`Style/QuotedSymbols ` doesn't follow `Style/StringLiterals`' `EnforcedStyle` if the latter is disabled

### ISSUE THREAD

**Issue by ccutrer** [rubocop/rubocop#10154]
In our codebase, we haven't enabled `Style/StringLiterals` yet because we're not quite ready to apply such a massive commit. But when we do, we will be using `EnforcedStyle: double_quotes`. However, we would like to start enforcing `Style/QuotedSymbols `. BUT, if we have in our .rubocop.yml:

```
Style/StringLiterals:
  EnforcedStyle: double_quotes
  Enabled: false # TODO: autocorrect
```

the QuotedSymbols cop does not see that configuration:

```
test.rb:4:1: C: [Correctable] Style/QuotedSymbols: Prefer single-quoted symbols when you don't need string interpolation or special symbols.
:"symbol-literal"
^^^^^^^^^^^^^^^^^
```

I know I can override the EnforcedStyle for Style/QuotedSymbols separately, but it's surprising that it doesn't inherit unless the cop is enabled.

## Expected behavior

Style/QuotedSymbols should respect Style/StringLiterals' EnforcedStyle param, regardless of the Enabled state of the latter.

## RuboCop version

```
1.21.0 (using Parser 3.0.2.0, rubocop-ast 1.11.0, running on ruby 2.7.2 x86_64-darwin20)
  - rubocop-rails 2.12.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.5.0
```

**Comment by dvandersluis** [comment]
Disabled cops are not used for configuration, but `Style/QuotedSymbols` has its own configuration if you don't have `StringLiterals` enabled to inherit from.

```yaml
Style/QuotedSymbols:
  EnforcedStyle: double_quotes
```

**Comment by ccutrer** [comment]
Is there anywhere that it's documented that disabled cops aren't used for configuration? Or a good reason why? Another one I've run into is Style/IfUnlessModifier will suggest that _all_ single-line-body if branches be converted to modifier form if Layout/LineLength is disabled. And it _doesn't_ have its own configuration parameter.

**Comment by dvandersluis** [comment]
There isn't a hard and fast rule, but it also doesn't make sense to have a disabled cop affect another cop IMO. Most cops do not even consider other cops, and cops are generally built to work in isolation of each other (with the exception being that many cops expect Layout cops to fix up layout after correction rather than reimplementing it in every cop).

[This discussion](https://github.com/rubocop/rubocop/discussions/10048) would be relevant to your `Style/IfUnlessModifier` question (or feel free to make a new issue about that specifically).

**Comment by ccutrer** [comment]
Thanks for the link to that discussion! And yes, it's a relatively rare occurrence that one cop follows the configuration of another. I can see arguments both ways for if the "source" cop for such following is enabled if the other parts follow. IMO it should at least be documented how it works (either in the overall configuration section of the docs, or on each cop that documents that it follows the configuration of another cop).

**Comment by dvandersluis** [comment]
I'm not against documenting it, but as I said, I don't know if it applies everywhere, or even an absolute list of cops that rely on other cops documentation.

I agree that the documentation for `QuotedSymbol` specifically can be improved however, so I'll do that. :+1:
